src/simulations/optimisation.py

```diff
@@ -29,6 +32,7 @@ def __init__(self, prm):
         self.save = prm["save"]
         self.paths = prm["paths"]
         self.manage_agg_power = prm["grd"]["manage_agg_power"]
+        self.record = Record(prm)
```

Comment:
no, do not create new record object, we have one overall record object for the whole run.